bundles/org.openhab.binding.livisismarthome/src/main/java/org/openhab/binding/livisismarthome/internal/handler/LivisiBridgeHandler.java

```diff
@@ -0,0 +1,936 @@
+        executeCommand(deviceId, CapabilityDTO.TYPE_ROLLERSHUTTERACTUATOR,
+                capabilityId -> client.setRollerShutterActuatorState(capabilityId, rollerShutterLevel));
+    }
+
+    /**
+     * Sends the command to start or stop moving the rollershutter (ISR2) in a specified direction
+     *
+     * @param deviceId device id
+     * @param action action
+     */
+    public void commandSetRollerShutterStop(final String deviceId, final ShutterActionType action) {
+        executeCommand(deviceId, CapabilityDTO.TYPE_ROLLERSHUTTERACTUATOR,
+                capabilityId -> client.setRollerShutterAction(capabilityId, action));
+    }
+
+    private void executeCommand(final String deviceId, final String capabilityType,
+            final CommandExecutor commandExecutor) {
+        if (deviceStructMan != null) {
+            try {
+                final Optional<String> capabilityId = deviceStructMan.getCapabilityId(deviceId, capabilityType);
+                if (capabilityId.isPresent()) {
+                    commandExecutor.executeCommand(capabilityId.get());
+                }
+            } catch (IOException | ApiException | AuthenticationException e) {
+                handleClientException(e);
+            }
+        }
+    }
+
+    ScheduledExecutorService getScheduler() {
+        return scheduler;
+    }
+
+    FullDeviceManager createFullDeviceManager(LivisiClient client) {
+        return new FullDeviceManager(client);
+    }
+
+    LivisiClient createClient(final OAuthClientService oAuthService) {
+        return new LivisiClient(bridgeConfiguration, oAuthService, new URLConnectionFactory());
+    }
+
+    /**
+     * Handles all Exceptions of the client communication. For minor "errors" like an already existing session, it
+     * returns true to inform the binding to continue running. In other cases it may e.g. schedule a reinitialization of
+     * the binding.
+     *
+     * @param e the Exception
+     * @return boolean true, if binding should continue.
+     */
+    private boolean handleClientException(final Exception e) {
+        boolean isReinitialize = true;
+        if (e instanceof SessionExistsException) {
+            logger.debug("Session already exists. Continuing...");
+            isReinitialize = false;
+        } else if (e instanceof InvalidActionTriggeredException) {
+            logger.debug("Error triggering action: {}", e.getMessage());
+            isReinitialize = false;
+        } else if (e instanceof RemoteAccessNotAllowedException) {
+            // Remote access not allowed (usually by IP address change)
+            logger.debug("Remote access not allowed. Dropping access token and reinitializing binding...");
+            refreshAccessToken();
+        } else if (e instanceof ControllerOfflineException) {
+            logger.debug("LIVISI SmartHome Controller is offline.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, e.getMessage());
```

Comment:
Hm, but this line sets bridge thing status. An API call responses that the controller (bridge device) is offline and this should be displayed at the bridge thing.

But it could also be that this error will not occur anymore, because the local API is now called (instead of the cloud API) and when the controller / bridge device would be offline, a network error would occur. I will check if this error can still occur.

Reply:
Sorry, I misread, you are setting bridge status here. The only thing wrong is the **BRIDGE_OFFLINE**. This is used for children of the bridge when they are offline because the bridge is offline. For the bridge itself, **COMMUNICATION_ERROR** would be more appropriate.

Reply:
solved